### PR TITLE
[AMDGPU] Implement moveToVALU for S_CSELECT_B64

### PR DESCRIPTION
**Pull request by jayfoad**
moveToVALU previously only handled S_CSELECT_B64 in the trivial case
where it was semantically equivalent to a copy. Implement the general
case using V_CNDMASK_B64_PSEUDO and implement post-RA expansion of
V_CNDMASK_B64_PSEUDO with immediate as well as register operands.

### CODE REVIEW

**Comment by jayfoad** [comment]
This should allow un-reverting #69703. The new test case `bug-cselect-b64.ll` would fail to compile with just #69703, but shows a slight codegen improvement when #69703 is applied on top of this patch.

**Comment by jayfoad** [comment]
Some of this was previously implemented as part of https://reviews.llvm.org/D109159 "[amdgpu] Enable selection of `s_cselect_b64`." but later reverted. @hliao2 @alex-t @piotrAMD

**Comment by piotrAMD** [comment]
Looks good as a natural extension of 32-bit select lowering. 

Just wondering if there is still something left to implement - I can see there were more test changes in [D109159](https://reviews.llvm.org/D109159) than here and #/69703. But maybe we already handle these cases.

**Comment by jayfoad** [comment]
Ping!

File: llvm/lib/Target/AMDGPU/SIISelLowering.cpp [hunk 1]
```diff
@@ -4737,20 +4737,42 @@ MachineBasicBlock *SITargetLowering::EmitInstrWithCustomInserter(
...
+        TRI->getSubRegisterClass(Src1RC, AMDGPU::sub1);
+
+    MachineOperand Src0Sub0 = TII->buildExtractSubRegOrImm(
+        MI, MRI, Src0, Src0RC, AMDGPU::sub0, Src0SubRC);
+    MachineOperand Src1Sub0 = TII->buildExtractSubRegOrImm(
+        MI, MRI, Src1, Src1RC, AMDGPU::sub0, Src1SubRC);
+
+    MachineOperand Src0Sub1 = TII->buildExtractSubRegOrImm(
+        MI, MRI, Src0, Src0RC, AMDGPU::sub1, Src0SubRC);
+    MachineOperand Src1Sub1 = TII->buildExtractSubRegOrImm(
```
Comment (rampitec) [inline llvm/lib/Target/AMDGPU/SIISelLowering.cpp:4759]: Wow! I didn't know it does exist. Seems useful.

**Review by rampitec — APPROVED** [review]
LGTM

**Comment by piotrAMD** [comment]
> > Just wondering if there is still something left to implement
> 
> Yes! The instruction selection parts of [D109159](https://reviews.llvm.org/D109159) still need to be unreverted. But I don't know exactly what problem caused them to be reverted in the first place, so I would like to proceed in small steps.

Thanks - sounds good.

**Review by piotrAMD — APPROVED** [review]

File: llvm/lib/Target/AMDGPU/SIISelLowering.cpp (hunk 1 above)
Comment (jayfoad) [reply llvm/lib/Target/AMDGPU/SIISelLowering.cpp:4759]: This is mostly copied from the handling of `S_ADD_U64_PSEUDO` and other 64-bit pseudos that get split.

**Pull request merged**